### PR TITLE
feat(components): update `tabs` to v9

### PR DESCRIPTION
**Pull request by ze-flo**
## Description

Updates `react-tabs` to v9.

## Details
- Refactors examples to use `Tabs`' subComponent properties
- Set `CodeExample`’s text color to `foreground.default` var

## V9 Migration checklist
- [x] Bump @zendeskgarden dependency
- [ ] ~~Update imports source~~
- [x] `Tabs` - Update examples to subcomponent properties 
- [ ] ~~`Field` - Update examples to subcomponent properties~~
- [ ] ~~`Grid` - Update examples to subcomponent properties~~
- [ ] ~~Update API breaking changes~~
- [ ] ~~MDX - Update docs to reflect migration changes~~
- [x] MDX - Update to subcomponent properties (handled [here](https://github.com/zendeskgarden/website/pull/621/files#diff-719cdc2fa3f81eeff582ab2efde23b365b3b521b3619c1cbfc1e2aff0db79e3dR77-R99))

Screenshot 2024-08-13 at 11 34 35 AM
Screenshot 2024-08-13 at 11 34 51 AM

## Checklist

- [ ] ~~:ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)~~
- [ ] ~~:black_nib: copy updates are approved (add the content strategist as a reviewer)~~
- [ ] ~~:link: considered opportunities for adding cross-reference URLs (grep for keywords)~~
- [ ] ~~:wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver~~
- [ ] ~~:memo: tested in Chrome, Firefox, Safari, Edge~~

### CODE REVIEW

File: src/components/MarkdownProvider/components/CodeExample/index.tsx [hunk 1]
```diff
@@ -128,6 +128,7 @@ export const CodeExample: React.FC<ICodeExampleProps> = ({ children, code }) =>
             border-top-right-radius: ${props => math(`${props.theme.borderRadii.md} - 1px`)};
             background-color: ${p => getColor({ theme: p.theme, variable: 'background.default' })};
             padding: ${p => p.theme.space.md};
+            color: ${p => getColor({ theme: p.theme, variable: 'foreground.default' })};
```
Comment (geotrev) [inline src/components/MarkdownProvider/components/CodeExample/index.tsx:131]: Why didn't I think of this before adding all these foreground styles to my PRs 🤣

**Review by geotrev — APPROVED** [review]
LGTM. I noticed the missing `border` in `Tabs.TabList`'s `transition` property. 😭

**Comment by ze-flo** [comment]
> LGTM. I noticed the missing `border` in `Tabs.TabList`'s `transition` property. 😭

The tab's border is inherited from color (via `currentcolor`) and `color` is a transition property.

https://github.com/zendeskgarden/react-components/blob/next/packages/tabs/src/styled/StyledTab.ts#L58

I think the transition treatment should work as intended. @geotrev Have you observed any inconsistencies?

**Comment by geotrev** [comment]
Ah, I figured it out. The `transition` needs to go on the `Tabs.TabList` container which has the overall subtle bottom border. Here it is working like I was trying to get:

https://github.com/user-attachments/assets/220adc5f-1118-44bd-b766-211b4f1e849d

Added `transition: border 0.25s ease-in-out` to `TabList`.

**Comment by ze-flo** [comment]
> Ah, I figured it out. The `transition` needs to go on the `Tabs.TabList` container which has the overall subtle bottom border. Here it is working like I was trying to get:
> 
>  Screen.Recording.2024-08-14.at.10.07.53.AM.mov 
> Added `transition: border 0.25s ease-in-out` to `TabList`.
> 
> Definitely not a blocker as I said, maybe better suited as a conversation with design first anyway.

Very nice! Thanks for diving into this. Filed a bug ticket.

**Review by jzempel — APPROVED** [review]

**Pull request merged**